pkg/resource/role/hooks.go

```diff
@@ -528,3 +567,41 @@ func (rm *resourceManager) removeTags(
 func decodeDocument(encoded string) (string, error) {
 	return url.QueryUnescape(encoded)
 }
+
+// sliceOrString defines a type that can be either a string or a slice.
+type sliceOrString []string
+
+// policyDocument represents a parsed AWS IAM policy document.
+type policyDocument struct {
+	Version    string       `json:"Version"`
+	Statements []*statement `json:"Statement"`
+}
+
+// statement is a single AWS IAM policy statement.
+type statement struct {
+	Effect     string                   `json:"Effect"`
+	Actions    sliceOrString            `json:"Action"`
+	Resources  sliceOrString            `json:"Resource,omitempty"`
+	Principals map[string]sliceOrString `json:"Principal,omitempty"`
+}
```

Comment:
Just learned about this today https://github.com/micahhausler/aws-iam-policy - it would be better to use a tested library...
/hold

Reply:
Just made the change.
/unhold